crates/api_models/src/payments.rs

```diff
@@ -114,6 +114,298 @@ pub struct CustomerDetailsResponse {
+    pub frm_metadata: Option<pii::SecretSerdeValue>,
+
+    /// Whether to perform external authentication (if applicable)
+    #[schema(value_type = External3dsAuthenticationRequest)]
+    #[serde(default)]
+    pub request_external_three_ds_authentication: common_enums::External3dsAuthenticationRequest,
+}
+
+#[cfg(feature = "v2")]
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize, ToSchema)]
+pub struct AmountDetails {
+    /// The payment amount. Amount for the payment in the lowest denomination of the currency, (i.e) in cents for USD denomination, in yen for JPY denomination etc. E.g., Pass 100 to charge $1.00 and 1 for 1¥ since ¥ is a zero-decimal currency. Read more about [the Decimal and Non-Decimal Currencies](https://github.com/juspay/hyperswitch/wiki/Decimal-and-Non%E2%80%90Decimal-Currencies)
+    #[schema(value_type = u64, example = 6540)]
+    #[serde(default, deserialize_with = "amount::deserialize")]
+    order_amount: Amount,
+    /// The currency of the order
+    #[schema(example = "USD", value_type = Currency)]
+    currency: common_enums::Currency,
+    /// The shipping cost of the order. This has to be collected from the merchant
+    shipping_cost: Option<MinorUnit>,
+    /// Tax details related to the order. This will be calculated by the external tax provider
+    tax_details: Option<TaxDetails>,
+    /// The action to whether calculate tax by calling external tax provider or not
+    #[serde(default)]
+    #[schema(value_type = TaxCalculationOverride)]
+    skip_external_tax_calculation: common_enums::TaxCalculationOverride,
+    /// The action to whether calculate surcharge or not
+    #[serde(default)]
+    #[schema(value_type = SurchargeCalculationOverride)]
+    skip_surcharge_calculation: common_enums::SurchargeCalculationOverride,
+    /// The surcharge amount to be added to the order, collected from the merchant
+    surcharge_amount: Option<MinorUnit>,
+    /// tax on surcharge amount
+    tax_on_surcharge: Option<MinorUnit>,
+}
+
+#[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct TaxDetails {
+    /// This is the tax related information that is calculated irrespective of any payment method.
+    /// This is calculated when the order is created with shipping details
+    pub default: Option<DefaultTax>,
+
+    /// This is the tax related information that is calculated based on the payment method
+    /// This is calculated when calling the /calculate_tax API
+    pub payment_method_type: Option<PaymentMethodTypeTax>,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct PaymentMethodTypeTax {
+    /// The order tax amount for the payment method type
+    pub order_tax_amount: MinorUnit,
+    /// The payment method type
+    #[schema(value_type = PaymentMethodType, example = "google_pay")]
+    pub pmt: common_enums::PaymentMethodType,
```

Comment:
Can we please avoid abbreviating things in API models? Can we name this as `payment_method_type`?

CC: @jarnura @Narayanbhat166

Reply:
will take this up in next PR